### PR TITLE
Add a licence data file

### PR DESCRIPTION
**Pull request by nahteb**
- Add a data file for licences 
- Add a test to validate the licences in the data file conform to our schema https://github.com/alphagov/data-standards-authority/blob/main/adr/0001-standards-catalogue-data-structure.md

### CODE REVIEW

File: content/standards/openreferraluk/index.html.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ status: review
 dateAdded: 2020-12-16
 dateUpdated: 2021-07-05
 classification: Service, Product, Application, System or Platform Design
+licence_id: 'CC-BY-SA-4.0'
```
Comment (jamietanna) [inline content/standards/openreferraluk/index.html.md:12]: Per a comment from Arnau, we should either be consistent (double quotes) or leave it off
```suggestion
licence_id: "CC-BY-SA-4.0"
```

File: standards-catalogue/.schema/licence.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    },
+    "url": {
+      "$comment": "A link to the licence",
+      "type": "string",
+      "pattern": "^https?://.*"
+    },
+    "type": {
+      "$comment": "The type of licence",
+      "type": "string"
+    }
```
Comment (jamietanna) [inline standards-catalogue/.schema/licence.json:25]: This should be an enum of values

**Review by jamietanna — CHANGES_REQUESTED** [review]
Couple of tweaks, but looks good :+1:

File: standards-catalogue/source/layouts/standards_table.erb [hunk 3]
```diff
@@ -16,6 +16,8 @@
             <td nowrap><strong><%= display_organisation(value) %></strong> </td>
           <% elsif key == 'status' %>
             <td nowrap><strong><%= display_status(:standard, value) %></strong> </td>
+          <% elsif key == 'licence_id' %>
+            <%= display_licence(value) %>
```
Comment (nahteb) [inline standards-catalogue/source/layouts/standards_table.erb:20]: Will remove all these if statements when I fix all the other frontmatter content

File: content/standards/openreferraluk/index.html.md (hunk 1 above)
Comment (nahteb) [reply content/standards/openreferraluk/index.html.md]: I've left it off all the others so removed the quotes from this one

**Review by jamietanna — APPROVED** [review]

**Review by jamietanna — CHANGES_REQUESTED** [review]
See below

File: standards-catalogue/data/licences.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  url: 'https://spdx.org/licenses/OGL-UK-3.0.html'
+CC-BY-SA-4.0:
+  name: Creative Commons Attribution Share Alike 4.0 International
+  type: Open standard
+  url: 'https://spdx.org/licenses/CC-BY-SA-4.0.html'
+proprietary:
+  name: Proprietary
+  type: Proprietary
+unknown:
+  name: Unkown
```
Comment (jamietanna) [inline standards-catalogue/data/licences.yml:17]:
```suggestion
  name: Unknown
```

File: standards-catalogue/Rakefile [hunk 5]
```diff
@@ -58,9 +58,19 @@ namespace :schema do
...
       raise unless errors.all?
     end
+
+    desc "Validate licences are schema compliant"
+    task :licences do
+      schema = schema("licence")
+      organisations = YAML.load_file("data/licences.yml")
+      errors = organisations.map do |f|
+        is_valid_schema(schema, f[1])
+      end
```
Comment (jamietanna) [inline standards-catalogue/Rakefile:68]:
```suggestion
      licenses = YAML.load_file("data/licences.yml")
      errors = licenses.map do |f|
        is_valid_schema(schema, f[1])
      end
```

File: standards-catalogue/Rakefile [hunk 6]
```diff
@@ -58,9 +58,19 @@ namespace :schema do
       end
       raise unless errors.all?
     end
+
+    desc "Validate licences are schema compliant"
+    task :licences do
+      schema = schema("licence")
+      organisations = YAML.load_file("data/licences.yml")
+      errors = organisations.map do |f|
+        is_valid_schema(schema, f[1])
```
Comment (jamietanna) [inline standards-catalogue/Rakefile:67]: I'm happy to raise this as a follow-up, so we can tweak the existing organisations code, or if you'd like you can do both here, so they'd be i.e.:

```suggestion
      errors = organisations.map do |_, organisation|
        is_valid_schema(schema, organisation)
```

File: standards-catalogue/config.rb [hunk 7]
```diff
@@ -90,6 +90,14 @@ def display_status(type, id)
     end
   end
 
+  def display_licence(id)
+    if data.licences[id].url
+      "<td nowrap><strong>#{data.licences[id].type} - <a href=#{data.licences[id].url}>#{data.licences[id].name}</a></strong></td>"
+    else
+      "<td nowrap><strong>#{data.licences[id].name}</strong></td>"
+    end
+  end
```
Comment (jamietanna) [inline standards-catalogue/config.rb:99]: To remove duplicated lookups (not necessarily costly complexity wise, but also has the risk of us leading to us likely forgetting to update a reference in the future)

```suggestion
    license = data.licences[id]
    if license.url
      "<td nowrap><strong>#{license.type} - <a href=#{license.url}>#{license.name}</a></strong></td>"
    else
      "<td nowrap><strong>#{license.name}</strong></td>"
    end
  end
```

File: standards-catalogue/Rakefile (hunk 6 above)
Comment (nahteb) [reply standards-catalogue/Rakefile]: Added another commit to address these https://github.com/alphagov/data-standards-authority/pull/126/commits/cc03ee1ff5fab704516818ed7bfec6060085bc79

**Review by jamietanna — CHANGES_REQUESTED** [review]
Last thing, sorry!

**Comment by nahteb** [comment]
> Last thing, sorry!

Can't see what the new requested change is sorry - think I've addressed the ones that were there?

**Review by jamietanna — APPROVED** [review]
Woops, looked at the old diff - it was that I was seeing `license` not `licence` :+1:

**Pull request merged**